### PR TITLE
Changed AWS Lambda IAM Authorization Example

### PR DESCRIPTION
**Pull request by renegoretzka**
_Issue #3623_

_Description of changes:_
Changed the part of the AWS Lambda function IAM Authorization to a simple, lightweight, and easy to understand.
This follows more like the API.graphql() method from aws-amplify/js approach.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/graphqlapi/graphql-from-node.mdx [hunk 1]
```diff
@@ -150,84 +147,60 @@ General information
 You can access the following resource attributes as environment variables from your Lambda function
 	API_<YOUR_API_NAME>_GRAPHQLAPIENDPOINTOUTPUT
 	API_<YOUR_API_NAME>_GRAPHQLAPIIDOUTPUT
-	API_<YOUR_API_NAME>_GRAPHQLAPIKEYOUTPUT
 ? Do you want to edit the local lambda function now? No
 ```
 
 The CLI will automatically configure the Lambda execution IAM role required by the Lambda function to call the GraphQL API. The following function will sign the request and use environment variables for the AppSync and Region that `amplify add function` created for you.
 
+Next, you need to install two npm packages for your AWS Lambda function. To do so, open your terminal in the `amplify/backend/function/<YOUR_FUNCTION_NAME>/src` directory in your newly created AWS Lambda function and run the command `npm install appsync-client graphql-tag`. Once the packages are installed, edit your index.js as the following:
```
Comment (evcodes) [inline src/fragments/lib/graphqlapi/graphql-from-node.mdx:155]: Next, you need to install two npm packages for your AWS Lambda function. To do so, open your terminal in the `amplify/backend/function/<YOUR_FUNCTION_NAME>/src` directory in your newly created AWS Lambda function and run the command `npm install appsync-client graphql-tag`. Once the packages are installed, add this your ```index.js```:

File: src/fragments/lib/graphqlapi/graphql-from-node.mdx [hunk 2]
```diff
@@ -150,84 +147,60 @@ General information
...
+      }
     };
+  } catch (error) {
+    console.log('Something went wrong in the handler:', error);
+  }
 };
 ```
 
-Finally you can define the GraphQL operation you're running, in this case the `createTodo` mutation, in a separate `query.js` file:
+Finally you can define the GraphQL operation you're running, in this case the `createTodo` mutation, in a separate `graphql.js` file:
```
Comment (evcodes) [inline src/fragments/lib/graphqlapi/graphql-from-node.mdx:194]: You can also define which GraphQL operation you're running. In this example, we are going to be creating a `createTodo` mutation within a separate `graphql.js` file.

**Review by evcodes — CHANGES_REQUESTED** [review]
Small grammatical nits but LGTM other than that. This dramatically simplifies Lambda authorizers and is the right thing to do from a security perspective.

**Comment by renegoretzka** [comment]
Thank you @evcodes! I applied the suggested changes!

File: src/fragments/lib/graphqlapi/graphql-from-node.mdx [hunk 3]
```diff
@@ -150,84 +147,60 @@ General information
...
-        statusCode: 200,
-        body: data
+      statusCode: 200,
+      body: JSON.stringify(data.createTodo),
+      headers: {
+        'Access-Control-Allow-Origin': '*'
+      }
     };
+  } catch (error) {
+    console.log('Something went wrong in the handler:', error);
```
Comment (sachscode) [inline src/fragments/lib/graphqlapi/graphql-from-node.mdx:189]: can we use console.error here ?

File: src/fragments/lib/graphqlapi/graphql-from-node.mdx [hunk 4]
```diff
@@ -150,84 +147,60 @@ General information
...
+const AppSyncClient = require('appsync-client').default;
+const gql = require('graphql-tag');
+const config = {
+  apiUrl: process.env.API_ < YOUR_API_NAME > _GRAPHQLAPIENDPOINTOUTPUT
+};
+const client = new AppSyncClient(config);
 
-exports.handler = async (event) => {
-    const req = new AWS.HttpRequest(appsyncUrl, region);
+const { createTodo } = require('./graphql.js');
```
Comment (sachscode) [inline src/fragments/lib/graphqlapi/graphql-from-node.mdx:165]:
```suggestion
const { createTodo } = require('./graphql.js'); /* GraphQL mutation */
```

File: src/fragments/lib/graphqlapi/graphql-from-node.mdx [hunk 5]
```diff
@@ -150,84 +147,60 @@ General information
...
-const https = require('https');
-const AWS = require("aws-sdk");
-const urlParse = require("url").URL;
-const appsyncUrl = process.env.API_<YOUR_API_NAME>_GRAPHQLAPIENDPOINTOUTPUT;
-const region = process.env.REGION;
-const endpoint = new urlParse(appsyncUrl).hostname.toString();
-const graphqlQuery = require('./query.js').mutation;
-const apiKey = process.env.API_<YOUR_API_NAME>_GRAPHQLAPIKEYOUTPUT;
+const AppSyncClient = require('appsync-client').default;
+const gql = require('graphql-tag');
```
Comment (sachscode) [inline src/fragments/lib/graphqlapi/graphql-from-node.mdx:159]:
```suggestion
const gql = require('graphql-tag'); /* To parse GraphQL strings into GraphQL AST */
```

**Review by sachscode — COMMENTED** [review]
LGTM, just a few comments

**Comment by mauerbac** [comment]
hey @renegoretzka - you able to finish this up?

**Comment by renegoretzka** [comment]
Yeah, I will take care of it. Thanks for the reminder!

**Review by katiegoines — APPROVED** [review]

**Comment by josefaidt** [comment]
Hey @renegoretzka :wave: sorry for the delay here, and quite honestly I updated this page not too, too long ago to use node-fetch and aws-sdk v3. Can you take a look at the merge conflicts and see how this change resonates with the updated doc? https://docs.amplify.aws/lib/graphqlapi/graphql-from-nodejs/q/platform/js/#iam-authorization

**Comment by renegoretzka** [comment]
Hey @josefaidt 👋  maybe we can drop this PR in favor of the solution you have provided :)

**Comment by katiegoines** [comment]
@renegoretzka - Closing this PR per your comment. Thank you!

**Pull request closed without merging**